src/pages/tBTC/Bridge/Minting/Timeline.tsx

```diff
@@ -0,0 +1,196 @@
+import { useLocation } from "react-router"
+
+const StyledBadge: FC<Omit<BadgeProps, "variant" | "mt" | "marginTop">> = (
+  props
+) => {
+  const { account } = useWeb3React()
+  return account ? null : (
+    <Badge
+      variant="subtle"
+      width="fit-content"
+      display="block"
+      mt={4}
+      {...props}
+    />
+  )
+}
+
+type TimelineItemType = {
+  id: DepositDetailsStep | MintingStep
+  label: string
+  description: ReactNode
+}
+
+const mintingItems: TimelineItemType[] = [
+  {
+    id: MintingStep.ProvideData,
+    label: "Deposit Address",
+    description: (
+      <>
+        Provide an ETH address and a BTC Recovery address to generate an unique
+        BTC deposit address.
+        <StyledBadge>ACTION OFF-CHAIN</StyledBadge>
+      </>
+    ),
+  },
+  {
+    id: MintingStep.Deposit,
+    label: "Make a BTC deposit",
+    description: (
+      <>
+        Send any amount lager than 0.01 BTC to this unique BTC Deposit Address.
+        The amount sent will be used to mint tBTC.
+        <StyledBadge>ACTION ON BITCOIN</StyledBadge>
+      </>
+    ),
+  },
+  {
+    id: MintingStep.InitiateMinting,
+    label: "Initiate minting",
+    description: (
+      <>
+        Minting tBTC does not require you to wait for the Bitcoin confirmations.
+        Sign an Ethereum transaction in your wallet and your tBTC will arrive in
+        around 1 to 3 hours.
+        <StyledBadge>ACTION ON ETHEREUM</StyledBadge>
+      </>
+    ),
+  },
+]
+
+const depositDetailsItems: TimelineItemType[] = [
+  {
+    id: DepositDetailsStep.BitcoinConfirmations,
+    label: "Bitcoin Confirmations",
+    description:
+      "Minting tBTC does not require you to wait for the Bitcoin confirmations. Sign an Ethereum transaction in your wallet and your tBTC will arrive in around 1 to 3 hours.",
+  },
+  {
+    id: DepositDetailsStep.MintingInitialized,
+    label: "Minter Check",
+    description: (
+      <>
+        Our Minters are inspecting the deposit. A Minter will initiate the token
+        issuance if the deposit is confirmed.
+        <br />
+        <br />A Minter is a software empowered by the Threshold DAO. They
+        provide security against malicious deposits.
+      </>
+    ),
+  },
+  {
+    id: DepositDetailsStep.GuardianCheck,
+    label: "Guardian Check",
+    description: (
+      <>
+        Our Guardians are inspecting the deposit. This is the second layer of
+        deposit validation.
+        <br />
+        <br />
+        Guardians are software empowered by the DAO. They can deny token
+        issuance, if the Bitcoin deposit is not confirmed.
+      </>
+    ),
+  },
+  {
+    id: DepositDetailsStep.Completed,
+    label: "Completing Deposit",
+    description: "The contract is sending and minting your tBTC tokens.",
+  },
+]
+
+type MintingTimelineProps = {
+  title: string
+} & BoxProps
+
+export const Timeline: FC<MintingTimelineProps> = ({ title, ...restProps }) => {
```

Comment:
I would extract `Timeline` and `TimelineItem` to a separate folder in `src/components` instead of keeping them in `src/pages/tBTC/Bridge/Minting`. It would be best if we replace our old `Timeline` component with this new one.

For this to happen we would have to slighty refactor the current `Timeline` component to make it more generic so that it doesn't contain any things related to tbtc. Then, the `Minting/Timelien` (or MintingTimeline) component would use the `Timeline` component and it's items we've just extracted. I would also add a `variant` property there and keep it in the context (for example) so that we know for sure that every `TimelineItem` has the same variant.

The reason why I would do that is because we might use `Timeline` in some other places in the future, for example for Unminting, as I assume we will implement it sooner or later. This will also make `Timeline` component more reusable in general.

Reply:
If you agree, then ideally we would also move the styles for it to a chakra theme.

Reply:
Looks great 👍